### PR TITLE
fix(docker): support non amd64 dockerize in setup containers

### PR DESCRIPTION
**Pull request by tonycsoka**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

This fixes an issue where the `amd64` binary version of `dockerize` was being used to start the `mysql-setup` and `elasticsearch-setup` containers, regardless of the architecture of image being being, causing `datahub docker quickstart` to fail on `arm64`.  Instead, this fix builds `dockerize` in a staged build, and uses the resultant binary in the final image.  Tried first by downloading the gzipped Source asset from the `dockerize` repo, but this did not include all needed files for a build, hence going for a git clone solution.

### CODE REVIEW

**Comment by pedro93** [comment]
Hello @tonycsoka 

First of all, thank you for making a contribution to DataHub! We appreciate folks who actively try to improve the project.

Second, for this PR to be accepted could you please update the PR title and description to match our guidelines? See here: https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format

In order to ensure your PR gets merged we would highly recommend adding a description to your PR that describes the problem you are trying to solve, how did you did it and any considerations reviewers and/or users of this feature may need to have.

**Comment by tonycsoka** [comment]
> Hello @tonycsoka
> 
> First of all, thank you for making a contribution to DataHub! We appreciate folks who actively try to improve the project.
> 
> Second, for this PR to be accepted could you please update the PR title and description to match our guidelines? See here: https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format
> 
> In order to ensure your PR gets merged we would highly recommend adding a description to your PR that describes the problem you are trying to solve, how did you did it and any considerations reviewers and/or users of this feature may need to have.

Thanks @pedro93, how's that for you.

File: docker/elasticsearch-setup/Dockerfile [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
+FROM golang:1.17-alpine3.15 AS binary
+ENV DOCKERIZE_VERSION 0.6.1
+WORKDIR /go/src/github.com/jwilder
+RUN apk --no-cache --update add openssl git tar curl \
+    && git clone https://github.com/jwilder/dockerize.git
+WORKDIR /go/src/github.com/jwilder/dockerize
+
+ENV GO111MODULE=on
+RUN go mod tidy
+RUN go install
```
Comment (szalai1) [inline docker/elasticsearch-setup/Dockerfile:15]: `go install github.com/jwilder/dockerize@latest`

might be simpler

**Comment by pedro93** [comment]
@tonycsoka did you try Peter's suggestion with `go install github.com/jwilder/dockerize@latest` rather than git cloning a repo?

File: docker/elasticsearch-setup/Dockerfile (hunk 1 above)
Comment (tonycsoka) [reply docker/elasticsearch-setup/Dockerfile:15]: sorted on the branch, not too familiar with the suggestion from @szalai1 as I'm a go virgin, but for now it'll grab the specified tag

**Comment by tonycsoka** [comment]
> @tonycsoka did you try Peter's suggestion with `go install github.com/jwilder/dockerize@latest` rather than git cloning a repo?

Not yet, there's a little issue in that v0.6.1 of dockerize used this system called glock for dependency management which I'm looking at

File: docker/elasticsearch-setup/Dockerfile (hunk 1 above)
Comment (tonycsoka) [reply docker/elasticsearch-setup/Dockerfile]: @pedro93 OK, hopefully that should be it, moved directly to `go install github.com/jwilder/dockerize@...` as per @szalai1

**Review by pedro93 — APPROVED** [review]
LGTM! Thank you for taking care of the review concerns. Once CI is green I think we can merge this.

**Comment by tonycsoka** [comment]
> LGTM! Thank you for taking care of the review concerns. Once CI is green I think we can merge this.

Great and have a fine weekend!

**Comment by tonycsoka** [comment]
> LGTM! Thank you for taking care of the review concerns. Once CI is green I think we can merge this.

@pedro93 Are those smoke tests a little delicate, can't see why this change would cause an issue and we have 2 different failures on separate CI runs?

**Comment by pedro93** [comment]
Yes our tests are a little flaky sometimes when big changes to the UI happen. CI was failing due to another issue that got fixed since. Merged your PR. Thank you for the contribution!

**Pull request merged**